File: packages/sdk-php/src/AccountsAPI.php
<?php

declare(strict_types=1);

namespace Melaya;

/**
 * Accounts API — user profile, credits, and GDPR data export.
 *
 * Maps to /api/v1/private/accounts/* and /api/v1/private/keys/:keyId.
 *
 * @example
 * ```php
 * $sdk = new Melaya\Melaya(apiKey: getenv('MK'));
 * $credits = $sdk->accounts->credits();
 * ```
 */
class AccountsAPI
{
    public function __construct(private readonly HttpClient $http) {}

    /**
     * Export all user data (GDPR Art 15/20).
     * Returns a JSON blob of all user data.
     */
    public function exportMyData(): array
    {
        return $this->http->post('/api/v1/private/accounts/export');
    }

    /**
     * Remove a stored CEX API key by its ID.
     *
     * @param string $keyId The key ID to remove (e.g. "BINANCEUSDM_0")
     */
    public function removeKey(string $keyId): array
    {
        return $this->http->delete('/api/v1/private/keys/' . rawurlencode($keyId));
    }

    /**
     * Update user display name, avatar, or settings.
     *
     * @param array $body Fields to update (e.g. ['displayName' => 'Alice'])
     */
    public function updateProfile(array $body): array
    {
        return $this->http->patch('/api/v1/private/accounts/profile', $body);
    }

    /** Return current credit balance and transaction history. */
    public function credits(): array
    {
        return $this->http->get('/api/v1/private/accounts/credits');
    }

    /** Return AI/LLM credit balance. */
    public function aiCredits(): array
    {
        return $this->http->get('/api/v1/private/accounts/credits/ai');
    }

    /** Return portfolio-ideas feature credit balance. */
    public function portfolioIdeasCredits(): array
    {
        return $this->http->get('/api/v1/private/accounts/credits/portfolio-ideas');
    }

    /** Return risk-monitoring feature credit balance. */
    public function riskMonitoringCredits(): array
    {
        return $this->http->get('/api/v1/private/accounts/credits/risk-monitoring');
    }
}

File: packages/sdk-java/src/main/java/org/melaya/AccountsAPI.java
package org.melaya;

import com.fasterxml.jackson.databind.JsonNode;

import java.util.Map;

/**
 * Accounts API — profile management, GDPR export, API-key removal, and
 * credit balance reads.
 *
 * <p>Maps to {@code /api/v1/private/accounts/*} and {@code /api/v1/private/keys/*}.
 *
 * <p>Note: the <em>trading</em> account reads (exchange API keys, usage, api-key status)
 * live in {@link AccountAPI}. This class covers the platform-account operations
 * exposed under the {@code accounts} tRPC router.
 */
public class AccountsAPI {

    private final HttpClient http;

    AccountsAPI(HttpClient http) {
        this.http = http;
    }

    /**
     * GDPR Art 15/20 data export. Returns a JSON blob of all user data.
     * The response may be large; cache the result if needed.
     */
    public JsonNode exportMyData() {
        return http.post("/api/v1/private/accounts/export", null);
    }

    /**
     * Remove a stored CEX API key by ID.
     *
     * @param keyId the ID of the key to remove (from {@link AccountAPI#keys()})
     */
    public JsonNode removeKey(String keyId) {
        return http.delete("/api/v1/private/keys/" + keyId, null);
    }

    /**
     * Update the caller's display name, avatar, or settings.
     *
     * @param body a map of fields to update (e.g. {@code displayName}, {@code avatar})
     */
    public JsonNode updateProfile(Map<String, Object> body) {
        return http.patch("/api/v1/private/accounts/profile", body);
    }

    /**
     * Return the current credit balance and transaction history.
     * Covers all credit types (AI, trading, portfolio-ideas, etc.).
     */
    public JsonNode credits() {
        return http.get("/api/v1/private/accounts/credits", null);
    }

    /** Return the AI/LLM credit balance. */
    public JsonNode aiCredits() {
        return http.get("/api/v1/private/accounts/credits/ai", null);
    }

    /** Return the portfolio-ideas feature credit balance. */
    public JsonNode portfolioIdeasCredits() {
        return http.get("/api/v1/private/accounts/credits/portfolio-ideas", null);
    }

    /** Return the risk-monitoring feature credit balance. */
    public JsonNode riskMonitoringCredits() {
        return http.get("/api/v1/private/accounts/credits/risk-monitoring", null);
    }
}
